pkg/controller/llmisvc/lifecycle_crud.go

```diff
@@ -48,13 +48,15 @@ func Delete[O client.Object, T client.Object](ctx context.Context, c clientWithR
 	typeLogLine := logLineForObject(expected)
 	ownerLogLine := logLineForObject(owner)
 
-	if !metav1.IsControlledBy(expected, owner) {
-		return fmt.Errorf("failed to delete %s %s/%s: it is not controlled by %s %s/%s",
-			typeLogLine,
-			expected.GetNamespace(), expected.GetName(),
-			ownerLogLine,
-			owner.GetNamespace(), owner.GetName(),
-		)
+	if len(expected.GetNamespace()) != 0 {
```

Comment:
@bartoszmajsak Is there a better way for cluster-scoped resources?

Reply:
it seems to make a full remote call, I'd not use it for now as the non-empty for namespace is quite reliable already?

Reply:
I looked at the code and I saw caching mentioned here https://github.com/kubernetes-sigs/controller-runtime/blob/e08f24a1e29fb84e4b059dd471b1c8b4928751e7/pkg/client/apiutil/apimachinery.go#L75-L77 and subsequently here: https://github.com/kubernetes-sigs/controller-runtime/blob/e08f24a1e29fb84e4b059dd471b1c8b4928751e7/pkg/client/apiutil/restmapper.go#L170-L172

Which would mean it's going to be cached as we are using shared client? Or am I missing something?